pipeline/lib/pipeline.ml

```diff
@@ -188,15 +188,17 @@ let process_pipeline ~(docker_config : Docker_config.t) ~conninfo
         |> let> api, repo = repo in
            Github.Api.refs api repo
       in
+      let default_branch = Github.Api.default_ref refs in
```

Comment:
I'm a little bit confused by this. Where is `default_branch` actually used?

I think the code below should be:

```
match key with
| `Ref branch where branch = default_branch -> pipeline ~head ~branch ()
```

Reply:
Yep you're right, I got confused while reasoning about default_refs